tests/suites/test_suite_gcm.function

```diff
@@ -233,6 +233,74 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void gcm_encrypt_and_tag_reuse_ctx(int cipher_id, data_t *key_str,
+                                   data_t *src_str, data_t *iv_str,
+                                   data_t *add_str, data_t *dst,
+                                   int tag_len_bits, data_t *tag,
+                                   int iterations,
+                                   int init_result)
+{
+    unsigned char output[128];
+    unsigned char tag_output[16];
+    mbedtls_gcm_context ctx;
+    size_t tag_len = tag_len_bits / 8;
+    size_t n1;
+    size_t n1_add;
+    int nloop;
+
+    mbedtls_gcm_init(&ctx);
+
+    memset(output, 0x00, 128);
+    memset(tag_output, 0x00, 16);
+
+    /* Allocate a buffer for random input data */
+    unsigned char rand_input[128];
+    memset(rand_input, 0x00, 128);
+
+    TEST_ASSERT(mbedtls_gcm_setkey(&ctx, cipher_id, key_str->x,
+                                   key_str->len * 8) == init_result);
+    if (init_result == 0) {
+        /* Re-Use context for n iterations with random data */
+        for (nloop = 0; nloop <= iterations; nloop += 1) {
+            (void) mbedtls_test_rnd_std_rand(NULL, (void *) &rand_input, 128);
+            TEST_ASSERT(mbedtls_gcm_crypt_and_tag(&ctx, MBEDTLS_GCM_ENCRYPT,
+                                                  128, iv_str->x, iv_str->len,
+                                                  add_str->x, add_str->len,
+                                                  rand_input, output,
+                                                  tag_len, tag_output) == 0);
+        }
+
+        /* Now test and verify with valid input data without clearing the ctx */
+        TEST_ASSERT(mbedtls_gcm_crypt_and_tag(&ctx, MBEDTLS_GCM_ENCRYPT,
+                                              src_str->len, iv_str->x,
+                                              iv_str->len, add_str->x,
+                                              add_str->len, src_str->x,
+                                              output, tag_len,
+                                              tag_output) == 0);
+
+        ASSERT_COMPARE(output, src_str->len, dst->x, dst->len);
+        ASSERT_COMPARE(tag_output, tag_len, tag->x, tag->len);
+
+        /* Use the Multipart Interface to verify integrity */
+        for (n1 = 0; n1 <= src_str->len; n1 += 1) {
```

Comment:
There's no point in checking multiple breakdowns here since the objective is to test context reuse, not the multipart interface.

Conversely, the verification should happen with a separate context (and could use the one-shot function for simplicity), since the objective is to test context reuse. If the context is corrupt, this could cause the verification phase to pass wrongly.